nav2_behavior_tree/include/nav2_behavior_tree/bt_action_node.hpp

```diff
@@ -272,18 +316,38 @@ class BtActionNode : public BT::ActionNodeBase
         }
       };
 
-    auto future_goal_handle = action_client_->async_send_goal(goal_, send_goal_options);
+    goal_handle_.reset();
+    future_goal_handle_ = action_client_->async_send_goal(goal_, send_goal_options);
+    time_goal_sent_ = node_->now();
+    time_elapsed_since_goal_sent_ = std::chrono::milliseconds(0);
+  }
+
+  /**
+   * @brief Function to check if the action server acknowledged a new goal
+   * @return boolean True if future_goal_handle_ returns SUCCESS, False otherwise
+   */
+  bool is_future_goal_handle_complete()
+  {
+    auto timeout = server_timeout_ > bt_loop_duration_ ? bt_loop_duration_ : server_timeout_;
+    auto result = rclcpp::spin_until_future_complete(node_, future_goal_handle_, timeout);
+
+    time_elapsed_since_goal_sent_ =
```

Comment:
Shouldn't this be computed very first in the `tick`? Also I'm looking at `timeout = ...` and there's no looking at elapsed if less than `server_timeout_` or `bt_loop_duration_` duration remains to spin only the remainder.  I believe you had that before using `remaining` time.